### DISCUSSION TITLE
docs/core/glog-json

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#108] [General]
# docs/core/glog-json

使用GoFrame框架中的glog组件以JSON格式输出日志，适合日志分析工具解析。您将学习如何通过map或struct参数实现JSON日志格式输出，以及结合gjson.MustEncode方法实现更复杂的JSON内容输出。

https://goframe.org/docs/core/glog-json

**Comment by pengguan06** [comment]
能否输出阅读性更高的json格式，类似于g.Dump()。
如
```json
{
"uid":100,
"name":"john"
}
```

**Reply by gqcn** [reply]
`g.DumpJson(xxx)`?

**Reply by pengguan06** [reply]
这个只能终端输出吧，我希望通过glog打印到log文件中

**Reply by gqcn** [reply]
`g.Dump*`方法都是终端输出，用于调试的。你描述的场景使用使用`json.Marshal`使用`indent`参数控制格式化，再输出到`log`。

**Comment by dengzhipenglove** [comment]
json格式输出可以这样，把所有字段都放到json里
` {"level":"info","msg":"xxxx","ts":"2025-01-01T01:01:01Z"} `

**Reply by shiqinfeng1** [reply]
好像不支持，只有内容可以json，扔给日志服务器处理就不方便，建议官方支持一下

**Reply by shiqinfeng1** [reply]
修正一下： 可以使用handler支持
